packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -907,6 +920,20 @@ class _BottomNavigationBarState extends State<BottomNavigationBar> with TickerPr
         widget.unselectedFontSize,
       );
 
+    final IconThemeData effectiveSelectedIconTheme =
+      _effectiveIconTheme(
+        widget.selectedIconTheme ?? bottomTheme.selectedIconTheme,
+        widget.selectedItemColor,
+        widget.selectedLabelStyle,
+      );
+
+    final IconThemeData effectiveUnselectedIconTheme =
+      _effectiveIconTheme(
+        widget.unselectedIconTheme ?? bottomTheme.unselectedIconTheme,
+        widget.unselectedItemColor,
+        widget.unselectedLabelStyle,
```

Comment:
Same comment here: why are we using the label style's color for the icon? It describes the text color, not the icon color.

Reply:
@darrenaustin Same logic here
```dart
    final IconThemeData effectiveUnselectedIconTheme =
      _effectiveIconTheme(
        widget.unselectedIconTheme ?? bottomTheme.unselectedIconTheme,
        widget.unselectedItemColor 
        ?? bottomTheme.unselectedItemColor 
        ?? themeData.unselectedWidgetColor
      );
```